fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/AzureProvider.java

```diff
@@ -0,0 +1,316 @@
+                    CountInputStreamReader counter = new CountInputStreamReader(in);
+                    BufferedReader reader = new BufferedReader(counter)) {
+
+                List<String> lines = reader.lines().collect(Collectors.toList());
+                int idx = 1;
+                for (String line : lines) {
+                    idx++;
+                    String compLine = previous + line;
+
+                    try (ByteArrayInputStream bais = new ByteArrayInputStream(compLine.getBytes())) {
+                        resources.add(FHIRParser.parser(Format.JSON).parse(bais));
+                        if (LOG.isLoggable(Level.FINE)) {
+                            LOG.fine("Azure Resource [R] " + compLine);
+                        }
+                        // There is no reason to reset compLine as we're now stopping this proccessing block.
+                        // previous, baos are all goign to go out of scope.
+                        if (window > 0) {
+                            // Doesn't need to be closed per JavaDocs
+                            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                            baos.write(line.getBytes());
+                            currentBytes += baos.size();
+                            complete = true;
+                            break;
+                        }
+                    } catch(FHIRParserException fpe) {
+                        // We're on the last line, so there might be more
+                        if (idx -1 == lines.size()) {
+                            previous.append(line);
+                        }
+                    }
+                }
+                if (!complete) {
+                    currentBytes += counter.getLength();
+                    window++;
+                }
+            } catch (IOException e) {
+                LOG.throwing("AzureProvider", "readResources", e);
+                LOG.severe("Problem accessing backend on Azure" + e.getMessage());
+                throw new FHIRException("Problem accessing backend Container or Blob on Azure", e);
+            }
+        }
+    }
+
+    /**
+     * CountInputStreamReader counts the size of the input.
+     */
+    private static class CountInputStreamReader extends InputStreamReader {
+        private long l = 0;
+
+        public CountInputStreamReader(InputStream in) {
+            super(in);
+        }
+
+        @Override
+        public int read(char[] cbuf, int offset, int length) throws IOException {
+            int r = super.read(cbuf, offset, length);
+            if (r != -1) {
+                l += r;
+            }
+            return r;
+        }
+
+        /**
+         * @return the length of the resources returned in the reader
+         */
+        public long getLength() {
+            return l;
+        }
+    }
+
+    @Override
+    public void pushOperationOutcomes() throws FHIRException {
```

Comment:
I guess this is an outstanding TODO?

Reply:
not yet there... :) this should be a draft.  I got it working, I'm just more in progress